storage/mediator.go

```diff
@@ -131,16 +133,34 @@ func (m *mediator) EnableFileOps() {
 	m.databaseFileSystemManager.Enable()
 }
 
+// Tick mediates the relationship between ticks and flushes/snapshots/cleanups.
+//
+// For example, the requirements to perform a flush are:
+// 		1) currentTime > blockStart.Add(blockSize).Add(bufferPast)
+// 		2) node is not bootstrapping (technically shard is not bootstrapping)
+// 		3) at least one complete tick has occurred since blockStart.Add(blockSize).Add(bufferPast)
+//		4) at least one complete tick has occurred since bootstrap completed (can be the same tick
+// 		   that satisfies condition #3)
+//
+// The mediator helps ensure conditions #3 and #4 are met by measuring the tickStart time and the shard bootstrap
+// states *before* the tick, and then propagating those values to downstream components so they can use that
+// information to make decisions about whether to flush / snapshot / run cleanups.
 func (m *mediator) Tick(runType runType, forceType forceType) error {
-	start := m.nowFn()
+	tickStart := m.nowFn()
+	dbBootstrapStates, err := m.databaseBootstrapStates()
```

Comment:
nit: could you rename this to `dbBootstrapStateAtTickStart`, makes it much clearer why you can use it the way you're using it.